src/Control/HTTPApplication.php

```diff
@@ -141,4 +152,85 @@ public function execute(HTTPRequest $request, callable $callback, $flush = false
+                '4.13.0',
+                'silverstripe/graphql 3.x is deprecated. Upgrade to 4.x instead.'
+                . ' See https://docs.silverstripe.org/en/4/upgrading/upgrading_to_graphql_4/',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // The alternate_base_folder config property is deprecated, but because it's
+        // always fetched it'll throw a deprecation warning whether you've set it or not.
+        if (Director::baseFolder() !== BASE_PATH) {
+            Deprecation::notice(
+                '4.13.0',
+                'Use of an alternate base folder is deprecated',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // The alternate_public_dir config property is deprecated, but because it's
+        // always fetched it'll throw a deprecation warning whether you've set it or not.
+        // There are also multiple mechanisms which can result in this bad configuration.
+        if (PUBLIC_DIR !== 'public' || Director::publicFolder() !== PUBLIC_DIR) {
+            Deprecation::notice(
+                '4.13.0',
+                'Use of a public webroot other than "public" is deprecated.'
+                . ' See https://docs.silverstripe.org/en/4/changelogs/4.1.0#public-folder/',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // This change of defaults has no other deprecation notice being emitted currently.
+        $project = new Module(BASE_PATH, BASE_PATH);
+        if (RESOURCES_DIR !== '_resources' && $project->getResourcesDir() !== RESOURCES_DIR) {
+            Deprecation::notice(
+                '4.13.0',
+                'The RESOURCES_DIR constant will change to "_resources" by default.'
+                . ' See https://docs.silverstripe.org/en/5/changelogs/5.0.0/#api-general',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // File resolution strategy defaults changed in CMS 4.4 - people may not have changed their
+        // config since then, so alert them their legacy strategy won't work in the future.
+        $publicResolutionStrat = Injector::inst()->get(FileResolutionStrategy::class . '.public');
+        $protectedResolutionStrat = Injector::inst()->get(FileResolutionStrategy::class . '.protected');
+        if (method_exists($publicResolutionStrat, 'getResolutionFileIDHelpers')) {
+            $hasDeprecatedStrategy = false;
+            foreach ($publicResolutionStrat->getResolutionFileIDHelpers() as $helper) {
+                if ($helper instanceof LegacyFileIDHelper) {
+                    $hasDeprecatedStrategy = true;
+                    break;
+                }
+            }
+
+            if (!$hasDeprecatedStrategy && method_exists($protectedResolutionStrat, 'getResolutionFileIDHelpers')) {
+                foreach ($protectedResolutionStrat->getResolutionFileIDHelpers() as $helper) {
+                    if ($helper instanceof NaturalFileIDHelper) {
```

Comment:
Don't just check for `!($helper instanceof HashFileIDHelper)` here because that will raise deprecation warnings for people doing their own crazy magic and having their own custom file resolution strategies. We're pretty much just trying to catch people who started a new project before 4.5 when installer first slapped in the new strategy config override.